pandas/tests/io/formats/test_style.py

```diff
@@ -87,28 +87,131 @@ def test_update_ctx_flatten_multi_and_trailing_semi(self):
+    @pytest.mark.parametrize("do_render", [True, False])
+    def test_deepcopy(self, do_changes, do_render):
+        # Updated in GH39708
+        # Change some defaults (to check later if they are copied)
+        if do_changes:
+            style = [{"selector": "th", "props": [("foo", "bar")]}]
+            self.styler.set_table_styles(style)
+            attributes = 'class="foo" data-bar'
+            self.styler.set_table_attributes(attributes)
+            self.styler.uuid_len += 1
+            self.styler.hidden_index = not self.styler.hidden_index
+            self.styler.hide_columns("A")
+            classes = pd.DataFrame(
+                [["favorite-val red", ""], [None, "blue my-val"]],
+                index=self.df.index,
+                columns=self.df.columns,
+            )
+            self.styler.set_td_classes(classes)
+            ttips = pd.DataFrame(
+                data=[["Favorite", ""], [np.nan, "my"]],
+                columns=self.df.columns,
+                index=self.df.index,
+            )
+            self.styler.set_tooltips(ttips)
+            self.styler.cell_ids = not self.styler.cell_ids
+            self.styler.format("{:.2%}")
+
+        if do_render:
+            self.styler.render()
 
-    def test_deepcopy(self):
         s2 = copy.deepcopy(self.styler)
+
+        # Check for non-identity
         assert self.styler is not s2
         assert self.styler.ctx is not s2.ctx
         assert self.styler._todo is not s2._todo
-
+        assert self.styler.hidden_columns is not s2.hidden_columns
+        assert self.styler.cell_context is not s2.cell_context
+        if do_changes:  # self.styler.table_style is not None
+            assert self.styler.table_styles is not s2.table_styles
+        if do_changes:  # self.styler.tooltips is not None
+            assert self.styler.tooltips is not s2.tooltips
+            assert self.styler.tooltips.tt_data is not s2.tooltips.tt_data
+
+        # Check for equality (and changes in original objects)
         self.styler._update_ctx(self.attrs)
         self.styler.highlight_max()
         assert self.styler.ctx != s2.ctx
         assert s2._todo == []
         assert self.styler._todo != s2._todo
+        assert self.styler.table_styles == s2.table_styles
+        assert self.styler.table_attributes == s2.table_attributes
+        assert self.styler.cell_ids == s2.cell_ids
+        assert self.styler.uuid_len == s2.uuid_len
+        assert self.styler.hidden_index == s2.hidden_index
+        assert self.styler.hidden_columns == s2.hidden_columns
+        assert self.styler.cell_context == s2.cell_context
+        if do_changes:  # self.styler.table_style is not None
+            self.styler.table_styles[0]["selector"] = "ti"
+            assert self.styler.table_styles != s2.table_styles
+        if do_changes:  # self.styler.tooltips is not None
+            tm.assert_frame_equal(self.styler.tooltips.tt_data, s2.tooltips.tt_data)
```

Comment:
possible to wrap the two test cases copy and deepcopy so that dont repeat so much code?